tests/core/math/test_expression.h

```diff
@@ -213,6 +213,15 @@ TEST_CASE("[Expression] Underscored numeric literals") {
 	CHECK_MESSAGE(
 			expression.parse("0xff_99_00") == OK,
 			"The expression should parse successfully.");
+	CHECK_MESSAGE(
+			expression.parse("0Xff_99_00") == OK,
+			"The expression should parse successfully.");
+	CHECK_MESSAGE(
+			expression.parse("0b10_11_00") == OK,
+			"The expression should parse successfully.");
+	CHECK_MESSAGE(
+			expression.parse("0X10_11_00") == OK,
```

Comment:
I think you meant to use a binary prefix here (even if hexadecimal is technically valid here):

```suggestion
			expression.parse("0B10_11_00") == OK,
```